packages/create-webstone-app/src/index.ts

```diff
@@ -1,6 +1,155 @@
-import { displayWelcome, displayNextSteps } from "./helpers";
-import { tasks } from "./tasks/index";
+import { CreateWebstoneOptions, WebstoneAppType } from "../types";
+import fs from "fs-extra";
+import path from "node:path";
+import { PackageJson } from "type-fest";
+import { appPackageJson, pluginPackageJson } from "./package";
+import merge from "ts-deepmerge";
+import { create } from "create-svelte";
 
-displayWelcome();
-const context = await tasks.run();
-displayNextSteps(context);
+export async function createWebstone(
+  cwd: string,
+  options: CreateWebstoneOptions,
+) {
+  const { type, extendCLI } = options;
+  const appName = getAppName(cwd, type);
+  createBaseApp(cwd, { type, appName });
+  updatePackageJSON(cwd, { type, extendCLI });
+  if (type === "plugin" && extendCLI) {
+    copyBuildScript(cwd);
+    copyCLIExtension(cwd);
+  }
+}
+
+function getRawAppName(cwd: string) {
+  let appName: string = cwd;
+  if (cwd === ".") {
+    appName = process.cwd().split("/").pop() || "webstone-project";
+  }
+  return appName;
+}
+
+function getAppName(cwd: string, type: WebstoneAppType) {
+  let appName = getRawAppName(cwd);
+  if (type === "plugin") {
+    appName = `webstone-plugin-${appName}`;
+    return toValidPackageName(appName);
+  }
+  return toValidPackageName(appName);
+}
+
+function toValidPackageName(name: string) {
+  return name
+    .trim()
+    .toLowerCase()
+    .replace(/\s+/g, "-")
+    .replace(/^[._]/, "")
```

Comment:
Should this be a global regex?

```suggestion
    .replace(/^[._]/g, "")
```

Reply:
Ah... just as I posted, I realized it starts with `^`, so the `/g` is not necessary. I'll let you confirm that thought though 😊

Reply:
I basically copied that function from `create-svelte` LOL